skydb.go

```diff
@@ -0,0 +1,102 @@
+package skynet
+
+import (
+	"encoding/hex"
+	"fmt"
+	"gitlab.com/NebulousLabs/errors"
+	"io"
+	"io/ioutil"
+	"math"
+	"os"
+	"strings"
+)
+
+// GetJSON gets the JSON object corresponding to the publicKey and dataKey.
+func (sc *SkynetClient) GetJSON(
+	publicKey string,
+	dataKey string,
+) (io.ReadCloser, error) {
+	entry, err := sc.GetEntry(publicKey, dataKey)
+	if err != nil {
+		return nil, errors.AddContext(err, "could not get entry")
+	}
+
+	skylink, err := hex.DecodeString(entry.Data)
+	if err != nil {
+		return nil, errors.New("could not decode data")
+	}
+
+	return sc.Download(string(skylink), DefaultDownloadOptions)
+}
+
+// SetJSON sets a JSON object at the registry entry corresponding to the publicKey and dataKey.
+func (sc *SkynetClient) SetJSON(
+	privateKey string,
+	dataKey string,
+	json io.Reader,
+	revision *uint64,
+) (err error) {
+	if revision == nil {
+		privateKeyBytes, err := hex.DecodeString(privateKey)
+		if err != nil {
+			return errors.AddContext(err, "could not decode privateKey")
+		}
+
+		publicKeyBytes, err := publicKeyFromPrivateKey(privateKeyBytes)
+		if err != nil {
+			return errors.AddContext(err, "could not get publicKey from privateKey")
+		}
+
+		entry, err := sc.GetEntry(hex.EncodeToString(publicKeyBytes), dataKey)
+		if err != nil {
+			return errors.AddContext(err, "could not get entry")
+		}
+
+		newRevision := entry.Revision + 1
+		revision = &newRevision
+
+		if newRevision > math.MaxUint64 {
+			return errors.New("current entry already has maximum allowed revision, could not update the entry")
+		}
+	}
+
+	tempFile, err := createTempFileFromJson(dataKey, json)
```

Comment:
I don't think it should be necessary to create an actual physical file in order to upload data. We can just upload the streamed data directly. I believe there is a generic upload function in `upload.go` that does this, but if you need help don't hesitate to ask.